rules/DeadCode/Rector/Assign/RemoveUnusedAssignVariableRector.php

```diff
@@ -165,4 +174,20 @@ private function isNestedAssign(Assign $assign): bool
         $parent = $assign->getAttribute(AttributeKey::PARENT_NODE);
         return $parent instanceof Assign;
     }
+
+    private function isInCompact(Assign $assign): bool
+    {
+        $variable = $assign->var;
+
+        if (! $variable instanceof Variable) {
+            return false;
+        }
```

Comment:
`$variable` will always a `Variable` as already checked before, you can use `@var` for it:

```suggestion
        /** @var Variable $variable */
        $variable = $assign->var;
```

Reply:
Thanks!